.github/workflows/tag-maintainers.yml

```diff
@@ -73,79 +73,22 @@ jobs:
-      - name: Check maintainer collaborator status
-        id: check-collaborators
+      # Manage reviewers
+      - name: Manage reviewers
         env:
           GH_TOKEN: ${{ steps.app-token.outputs.token || secrets.GITHUB_TOKEN }}
+          OWNER: ${{ github.repository_owner }}
+          REPO: ${{ github.event.repository.name }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          PR_AUTHOR: ${{ github.event.pull_request.user.login }}
           MAINTAINERS: ${{ steps.extract-maintainers.outputs.maintainers }}
-          USERS_TO_EXCLUDE: ${{ steps.current-reviewers.outputs.users_to_exclude }}
-          REPO: "${{ github.repository }}"
+          CHANGED_FILES: ${{ steps.changed-files.outputs.changed_files }}
         run: |
-          NEW_REVIEWERS=()
-
-          # If there are no maintainers, exit early.
-          if [[ -z "$MAINTAINERS" ]]; then
-            echo "No maintainers to check."
-            echo "new_reviewers=" >> "$GITHUB_OUTPUT"
-            exit 0
-          fi
-
-          for MAINTAINER in $MAINTAINERS; do
-            if echo "$USERS_TO_EXCLUDE" | grep -q -w "$MAINTAINER"; then
-              echo "$MAINTAINER is already involved in the review, skipping."
-              continue
-            fi
-
-            echo "Checking if $MAINTAINER is a collaborator..."
-            if gh api "/repos/$REPO/collaborators/$MAINTAINER" --silent; then
-              echo "User $MAINTAINER is a collaborator, adding to new reviewers list."
-              NEW_REVIEWERS+=("$MAINTAINER")
-            else
-              echo "User $MAINTAINER is not a repository collaborator, skipping."
-            fi
-          done
-
-          NEW_REVIEWERS_LIST=$(printf "%s " "${NEW_REVIEWERS[@]}")
-          echo "new_reviewers=${NEW_REVIEWERS_LIST% }" >> "$GITHUB_OUTPUT"
-          echo "New reviewers to add: ${NEW_REVIEWERS_LIST% }"
-
-      # Add the new, filtered list of maintainers as reviewers to the PR.
-      - name: Add new reviewers
-        env:
-          GH_TOKEN: ${{ steps.app-token.outputs.token || secrets.GITHUB_TOKEN }}
-          NEW_REVIEWERS: ${{ steps.check-collaborators.outputs.new_reviewers }}
-          PR_NUM: ${{ github.event.pull_request.number }}
-        run: |
-          if [[ -n "$NEW_REVIEWERS" ]]; then
-            REVIEWERS_CSV=$(echo "$NEW_REVIEWERS" | tr ' ' ',')
-            echo "Requesting reviews from: $REVIEWERS_CSV"
-            gh pr edit "$PR_NUM" --add-reviewer "$REVIEWERS_CSV"
-          else
-            echo "No new reviewers to add."
-          fi
+          ./ci/tag-maintainers/manage-reviewers.py \
+            --owner "$OWNER" \
+            --repo "$REPO" \
+            --pr-number "$PR_NUMBER" \
+            --pr-author "$PR_AUTHOR" \
+            --current-maintainers "$MAINTAINERS" \
+            --changed-files "$CHANGED_FILES" \
+            --bot-user-name "${{ steps.app-token.outputs.app-slug }}"
```

Comment:
_Please_ use an env var 😭

Reply:
Also, when defining the var, we'll need `|| 'github-actions'` in case there is no app setup in the fork.